File: 14_FMDBDemo/FMDBDemo_oc/FMDBDemo_oc/View/HomeCell.m
//
//  HomeCell.m
//  FMDBDemo_oc
//
//  Created by sunke on 2020/9/19.
//  Copyright © 2020 KentSun. All rights reserved.
//

#import "HomeCell.h"
#import "User.h"

@interface HomeCell()
@property (weak, nonatomic) IBOutlet UILabel *nameLabel;
@property (weak, nonatomic) IBOutlet UILabel *timeLabel;
@property (weak, nonatomic) IBOutlet UILabel *contentLabel;

@end

@implementation HomeCell

- (void)setUser:(User *)user {
    _user = user;
    self.nameLabel.text = user.name;
    self.timeLabel.text = user.detail.created_at;
    self.contentLabel.text = user.detail.text;
}
@end

File: 14_FMDBDemo/FMDBDemo_swift_error/FMDBDemo_swift/View/HomeCell.swift
//
//  HomeCell.swift
//  FMDBDemo_swift
//
//  Created by sunke on 2020/9/19.
//  Copyright © 2020 KentSun. All rights reserved.
//

import UIKit

class HomeCell: UITableViewCell {
    @IBOutlet weak var nameLabel: UILabel!
    @IBOutlet weak var timeLabel: UILabel!
    @IBOutlet weak var contentLabel: UILabel!

    var user: User? {
        didSet {
            guard let user = user else {
                return
            }
            
            nameLabel.text = user.name
            guard let detail = user.detail else {
                return
            }
            
            timeLabel.text = detail.created_at
            contentLabel.text = detail.text
        }
    }

}
